doc/rst/lib_spi.rst

```diff
@@ -770,8 +770,122 @@ Throughput for SPI slave versus mode and MOSI usage is shown in the following ta
+
+.. _spi_master_example:
+
+.. uml::
+   :width: 60%
+   :caption: Task diagram of SPI master example
+
+   @startuml
+   left to right direction
+   circle app
+   circle SPI_master
+   app --> SPI_master : spi_master_if\nor\nspi_master_async_if
+   @enduml
+
+Declaring ports
+...............
+
+The SPI library connects to external pins via xCORE ports. In
+``main.xc`` these are declared as variables of type ``port`` at the
+start of the file:
+
+.. literalinclude:: ../../examples/app_spi_master/src/main.xc
+   :language: c
+   :start-at: p_sclk
+   :end-at: p_mosi
+
+.. note::
+
+  The slave select declaration is for a mulit-bit port. The pin in this
+  port that will be used as SPI SS set by ``spi.set_ss_port_bit(0, 1);``
+
+How the ports (e.g. ``XS1_PORT_1I``) relate to external pins will
+depend on the exact device being used. See the device datasheet for details.
+
+|newpage|
+
+The application main() function
+...............................
+
+Below is the source code for the main function of this application,
+which is taken from the source file ``main.xc``
+
+.. literalinclude:: ../../examples/app_spi_master/src/main.xc
+    :language: c
+    :start-after: // SPI async main
+    :end-before: // end async main
+
+.. literalinclude:: ../../examples/app_spi_master/src/main.xc
+    :language: c
+    :start-after: // SPI sync main
+    :end-before: // end sync main
+
+Looking at this in more detail you can see the following:
+
+  - The par functionality describes running two separate tasks in parallel
+
+  - The ``spi_master`` or ``spi_master_async`` task drives the SPI bus and takes the ports it
+    will use as arguments.
+
+  - The ``app`` or ``app_async`` task communicates to the SPI master task via the
+    shared interface argument ``i_spi`` or ``i_spi_async``. This is an array since the
+    SPI master task could connect to many other tasks (clients) in parallel.
+
+The app() function
+..................
+
+The ``app`` function uses its interface connection to the SPI master
+task to perform SPI transactions. It performs two transactions (each
+transaction will assert the slave select line, transfer some data and then
+de-assert the slave select line). The functions in the SPI master
+interface can be found in the SPI library user guide.
+
+.. literalinclude:: ../../examples/app_spi_master/src/main.xc
+    :language: c
+    :start-at: void app
+    :end-at: spi.shutdown
+
+.. literalinclude:: ../../examples/app_spi_master/src/main.xc
+    :language: c
+    :start-at: void async_app
+    :end-at: spi.shutdown
+
+
+.. note::
+
+    When ``begin_transaction`` is called the SPI device selected is determined
+    by the first argument. In this case it is ``0``.
+    This is the method that is used to communiate with multiple SPI
+    slave devices. The speed and mode of the SPI protocol is also set at
+    in the ``begin_transaction`` call.
+
+|newpage|
+
+
+
 Building
-========
+........
```

Comment:
Nab updated text from the template: https://github.com/xmosnotes/an0xxxx/blob/develop/doc/rst/an0xxxx.rst#example-application